release-notes.txt

```diff
@@ -1,3 +1,775 @@
+    * syscalls_default: update heap_stats for multiple heaps
+* sys/puf_sram: counter based seed after soft reset
+* sys/phydat: improvement of phydat_dump including test application
+* sys/shell:
+    + add rtt command
+    * correctly detect and handle long lines
++ sys/shell_commands: + provide command to print version
+* sys/spp: randomize canary value on each build
+* sys/suit:
+    * cleanup of TinyCBOR to NanoCBOR refactor
+    * coap: make use of exposed tree handler function
+    * update to draft-ietf-v3
++ sys/uri_parser: initial import of a minimal and non-destructive URI parsing
+* sys/usb: use default VID/PID for RIOT-included peripherals
+* sys/xtimer:
+    * set now pointer correctly in _update_short_timers() loop
+    + add xtimer_set_timeout_flag64()
+    * xtimer_mutex_lock_timeout fix with short timeout
++ sys/ztimer: initial import
+
+Networking
+----------
+* net/coap:
+    * deprecate gcoap_add_qstring() and update uses
+    + Packet API function to add Uri-Query option
++ net/dhcpv6: initial implementation of a client (with IA_PD support)
+* net/dns: fix DNS resolution in ping6
+* net/gcoap:
+    + add canonical uri query function names
+    + allow proxied client requests
+    * use sock_async and events
++ net/gnrc_dhcpv6_client_6lbr: initial import of a 6LBR DHCPv6 client
++ net/gnrc_ipv6_ext_frag: initial import of statistics module
+* net/gnrc_ipv6_nib:
+    * do not ignore PL for route
+    * fix border router with DNS & Context Compression
+    * only route to prefix list entry if on-link
+* net/gnrc_lorawan:
+    * fix the handling of downlink frames without payload
+    * remove netdev layer from MAC
+* net/gnrc_netif:
+    + add gnrc_netif_send and gnrc_netif_highlander functions
+    * implementation of dynamic GNRC_NETIF_NUMOF approach
+* net/gnrc_nettype: deprecate IOVEC type
+* net/gnrc_pktbuf: deprecate gnrc_pktbuf_replace_snip()
++ net/gnrc_sixlowpan_iphc: add support for IPv6 extension header compression
++ net/gnrc_sock: provide implementation for `sock_*_recv_buf()`
+* net/gnrc_sock_udp: choose random ephemeral port
+- net/gnrc_tftp: remove module
+* net/gnrc_[uhcp,dhcpv6_client_6lbr]: disable Router Advertisements on upstream
+  interface
+* net/gnrc_uhcpc:
+    * ensure compression context is managed by the ABR
+    * only configure 6Lo-ND features if wireless-interface is 6LN
+    * update compression context with new prefix
+* net/nanocoap:
+    + add coap_opt_add_uquery2() with parameter key value length
+    + add convenience function for adding path elements
+    * add uquery improvements
+    * make separate tree handling function
+    - remove obsolete functions for addition of Uri-Query option
++ net/netdev: add netdev_trigger_event_isr() function
+- net/nhdp: remove module
++ net/sock:
+    + amend API to iterate over stack-internal buffer chunks
+    + amend with zero-copy receive functions
+    * async: add optional callback argument
+    * async_event: fix race-condition
```

Comment:
After looking at it again, I actually would remove the last point (its a bug fix, not a feature) and then reverse the whole list. So add optional callback argument first ,then zero copy, then iteration